src/Microsoft.ML.TimeSeries/SsaSpikeDetector.cs

```diff
@@ -162,7 +162,7 @@ internal SsaSpikeDetector(IHostEnvironment env, ModelLoadContext ctx)
             InternalTransform.Host.CheckDecode(InternalTransform.IsAdaptive == false);
         }
 
-        public override void Save(ModelSaveContext ctx)
+        private protected override void SaveModel(ModelSaveContext ctx)
```

Comment:
>private protected override void SaveModel [](start = 8, length = 41)

Ahh, thank you very much for finding more of these!!

Reply:
Actually wait, these are already internal, so probably not necessary... still, eh... I don't know.

---
In reply to: [258775986](https://github.com/dotnet/machinelearning/pull/2673#discussion_r258775986) [](ancestors = 258775986)

Reply:
I think the containing classes are public. The classes in which I made the implementation of `ICanSaveModel,Save` explicit derive from  `SsaAnomalyDetectionBaseWrapper` or `IidAnomalyDetectionBaseWrapper` which are both public. 
There is just one that does not, and that's the Mapper class in `SequentialAnomalyDetectionTransformBase` which is internal. This was the only instance where a mapper class does not have explicit implementation of the `Save` method. In the previous PR, we decided the that despite this being an internal class it was beneficial to do an explicit implementation for consistency as it implements `ICanSaveModel`. So all mappers elsewhere have an explicit implementation of that method. 

---
In reply to: [258776058](https://github.com/dotnet/machinelearning/pull/2673#discussion_r258776058) [](ancestors = 258776058,258775986)